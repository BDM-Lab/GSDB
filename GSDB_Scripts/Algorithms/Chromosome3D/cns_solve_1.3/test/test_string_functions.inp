
evaluate ( $abc = "abcdefghijklmnopqrstuvwxyz" )
evaluate ( $1 = len($abc) )
evaluate ( $1 = len("123") )
evaluate ( $1 = substr($abc,15,19) )
evaluate ( $1 = substr($abc,-15,19) )
evaluate ( $1 = substr($abc,-15,-1) )
evaluate ( $1 = substr($abc,22,99) )
evaluate ( $1 = index($abc,"x") )
evaluate ( $1 = index($abc+$abc,"x") )
evaluate ( $1 = index($abc,"9") )
evaluate ( $1 = rindex($abc,"x") )
evaluate ( $1 = rindex($abc+$abc,"x") )
evaluate ( $1 = rindex($abc,"9") )
evaluate ( $1 = "d" * 20 )
evaluate ( $1 = adjustl("   xyz   ") )
evaluate ( $1 = trim(adjustl("   xyz   ")) )
evaluate ( $1 = trim("   xyz   ") )
evaluate ( $1 = match($abc,"*") )
evaluate ( $1 = trim(adjustl(format("F10.2",1/3))) )
evaluate ( $1 = trim(adjustl(format("F10.2",1/3))) )
evaluate ( $1 = decode(trim(adjustl(format("F10.2",1/3)))) * 4 )
evaluate ( $1 = match($abc,"*bcd*") )
evaluate ( $1 = match($abc,"*bc#*") )
evaluate ( $1 = match($abc,"*bc%*") )
evaluate ( $1 = match($abc,"*bc+*") )
! Test for wildcard escapes
evaluate ( $1 = match("*#%+","*#%+") )

evaluate ( $1 = match("*#%+","\*\#\%\+") )
evaluate ( $1 = match("abc*", "abc*" ) )
evaluate ( $1 = match("abcd", "abc*" ) ) 
evaluate ( $1 = match("abc*", "abc\*") )
evaluate ( $1 = match("abcd", "abc\*" ) )

eval($t= "15:49:00.3" )   ! from $time
eval($start=(decode(substr($t,1,2)) * 60
            + decode(substr($t,4,5))) * 60 + decode(substr($t,7,10)))

display $start

stop
