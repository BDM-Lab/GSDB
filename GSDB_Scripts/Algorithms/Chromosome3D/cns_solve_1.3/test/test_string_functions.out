          ============================================================
          |                                                          |
          |            Crystallography & NMR System (CNS)            |
          |                         CNSsolve                         |
          |                                                          |
          ============================================================
           Version: 1.3
           Status: General release
          ============================================================
           Written by: A.T.Brunger, P.D.Adams, G.M.Clore, W.L.DeLano,
                       P.Gros, R.W.Grosse-Kunstleve,J.-S.Jiang,J.M.Krahn,
                       J.Kuszewski, M.Nilges, N.S.Pannu, R.J.Read,
                       L.M.Rice, G.F.Schroeder, T.Simonson, G.L.Warren.
           Copyright (c) 1997-2010 Yale University
          ============================================================
           Running on machine: lewis4-r710-login-node223 (x86_64/Linux,64-bit)
                               with 16 threads
           Program started by: oeow39
           Program started at: 16:08:09 on 05-Jul-2018
          ============================================================

 FFT3C: Using FFTPACK4.1

 CNSsolve> 
 CNSsolve>evaluate ( $abc = "abcdefghijklmnopqrstuvwxyz" ) 
 EVALUATE: symbol $ABC set to "abcdefghijklmnopqrstuvwxyz" (string)
 CNSsolve>evaluate ( $1 = len($abc) ) 
 EVALUATE: symbol $1 set to    26.0000     (real)
 CNSsolve>evaluate ( $1 = len("123") ) 
 EVALUATE: symbol $1 set to    3.00000     (real)
 CNSsolve>evaluate ( $1 = substr($abc,15,19) ) 
 EVALUATE: symbol $1 set to "opqrs" (string)
 CNSsolve>evaluate ( $1 = substr($abc,-15,19) ) 
 EVALUATE: symbol $1 set to "lmnopqrs" (string)
 CNSsolve>evaluate ( $1 = substr($abc,-15,-1) ) 
 EVALUATE: symbol $1 set to "lmnopqrstuvwxyz" (string)
 CNSsolve>evaluate ( $1 = substr($abc,22,99) ) 
 EVALUATE: symbol $1 set to "vwxyz" (string)
 CNSsolve>evaluate ( $1 = index($abc,"x") ) 
 EVALUATE: symbol $1 set to    24.0000     (real)
 CNSsolve>evaluate ( $1 = index($abc+$abc,"x") ) 
 EVALUATE: symbol $1 set to    24.0000     (real)
 CNSsolve>evaluate ( $1 = index($abc,"9") ) 
 EVALUATE: symbol $1 set to    0.00000     (real)
 CNSsolve>evaluate ( $1 = rindex($abc,"x") ) 
 EVALUATE: symbol $1 set to    24.0000     (real)
 CNSsolve>evaluate ( $1 = rindex($abc+$abc,"x") ) 
 EVALUATE: symbol $1 set to    50.0000     (real)
 CNSsolve>evaluate ( $1 = rindex($abc,"9") ) 
 EVALUATE: symbol $1 set to    0.00000     (real)
 CNSsolve>evaluate ( $1 = "d" * 20 ) 
 EVALUATE: symbol $1 set to "dddddddddddddddddddd" (string)
 CNSsolve>evaluate ( $1 = adjustl("   xyz   ") ) 
 EVALUATE: symbol $1 set to "xyz      " (string)
 CNSsolve>evaluate ( $1 = trim(adjustl("   xyz   ")) ) 
 EVALUATE: symbol $1 set to "xyz" (string)
 CNSsolve>evaluate ( $1 = trim("   xyz   ") ) 
 EVALUATE: symbol $1 set to "   xyz" (string)
 CNSsolve>evaluate ( $1 = match($abc,"*") ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = trim(adjustl(format("F10.2",1/3))) ) 
 EVALUATE: symbol $1 set to "0.33" (string)
 CNSsolve>evaluate ( $1 = trim(adjustl(format("F10.2",1/3))) ) 
 EVALUATE: symbol $1 set to "0.33" (string)
 CNSsolve>evaluate ( $1 = decode(trim(adjustl(format("F10.2",1/3)))) * 4 ) 
 EVALUATE: symbol $1 set to    1.32000     (real)
 CNSsolve>evaluate ( $1 = match($abc,"*bcd*") ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match($abc,"*bc#*") ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match($abc,"*bc%*") ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match($abc,"*bc+*") ) 
 EVALUATE: symbol $1 set to FALSE (logical)
 CNSsolve>! Test for wildcard escapes 
 CNSsolve>evaluate ( $1 = match("*#%+","*#%+") ) 
 EVALUATE: symbol $1 set to FALSE (logical)
 CNSsolve> 
 CNSsolve>evaluate ( $1 = match("*#%+","\*\#\%\+") ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match("abc*", "abc*" ) ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match("abcd", "abc*" ) ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match("abc*", "abc\*") ) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ( $1 = match("abcd", "abc\*" ) ) 
 EVALUATE: symbol $1 set to FALSE (logical)
 CNSsolve> 
 CNSsolve>eval($t= "15:49:00.3" )   ! from $time 
 EVALUATE: symbol $T set to "15:49:00.3" (string)
 CNSsolve>eval($start=(decode(substr($t,1,2)) * 60 
 Expression=            + decode(substr($t,4,5))) * 60 + decode(substr($t,7,10))) 
 EVALUATE: symbol $START set to    56940.3     (real)
 CNSsolve> 
 CNSsolve>display $start 
56940.3
 CNSsolve> 
 CNSsolve>stop 
          ============================================================
           Maximum dynamic memory allocation:     8201488 bytes
           Maximum dynamic memory overhead:           208 bytes
           Program started at: 16:08:09 on 05-Jul-2018
           Program stopped at: 16:08:09 on 05-Jul-2018
           CPU time used:       0.0042 seconds
          ============================================================
